### ISSUE TITLE
Atom 0.29.16 update fails

### ISSUE THREAD

**Issue by afrz** [Glavin001/atom-beautify#1434]
# Description

Installation/update failure of atom-beautify 0.29.16 on Windows 7 x64.

# Steps to Reproduce

1. Try to install 0.29.16 (through Atom GUI or apm CLI)
2. Installing “atom-beautify@0.29.16” failed.

# Debug

[https://gist.github.com/afrz/046daf8f44f10d53d3bbd30d052cde04](https://gist.github.com/afrz/046daf8f44f10d53d3bbd30d052cde04)

``'D:\Program' n'est pas reconnu en tant que commande interne``

It seems that script **build-options** can not find a file located in **'Program Files'** directory (which contains a space character).

# Checklist

I have:
- [x] Tried uninstalling and reinstalling Atom Beautify to ensure it installed properly
- [x] Reloaded (or restarted) Atom to ensure it is not a caching issue
- [x] Searched through existing Atom Beautify Issues at https://github.com/Glavin001/atom-beautify/issues so I know this is not a duplicate issue
- [x] Filled out the Input, Expected, and Actual sections above or have edited/removed them in a way that fully describes the issue.
- [x] Generated debugging information and added link for `debug.md` Gist to this issue

**Comment by rejhgadellaabsl** [comment]
+1. With me the problem seems to be the space in my Windows user name. (I know, spaces are evil but when you use Windows or Mac, they slip in easier..)

I did update nodejs/npm to the latest stable version (6.9.2 and 3.10.9 respectively) as suggested just to rule that out.

```
> atom-beautify@0.29.16 postinstall C:\Users\RODERI~1\AppData\Local\Temp\apm-install-dir-11702-6348-11bypu1\node_modules\atom-beautify
> npm run build-options

C:\Users\RODERI~1\AppData\Local\Temp\apm-install-dir-11702-6348-11bypu1
`-- (empty)


npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN deprecated minimatch@0.2.12: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.0: use uuid module instead
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
'C:\Users\Roderick' is not recognized as an internal or external command,
operable program or batch file.
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\RODERI~1\AppData\Local\Temp\apm-install-dir-11702-6348-11bypu1\package.json'
npm WARN apm-install-dir-11702-6348-11bypu1 No description
npm WARN apm-install-dir-11702-6348-11bypu1 No repository field.
npm WARN apm-install-dir-11702-6348-11bypu1 No README data
npm WARN apm-install-dir-11702-6348-11bypu1 No license field.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Users\\Roderick Gadellaa\\AppData\\Local\\atom\\app-1.12.7\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Roderick Gadellaa\\AppData\\Local\\atom\\app-1.12.7\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Roderick Gadellaa\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Roderick Gadellaa\\.atom\\.apmrc" "install" "C:\\Users\\RODERI~1\\AppData\\Local\\Temp\\d-11702-6348-x49ncd\\package.tgz" "--runtime=electron" "--target=1.3.13" "--arch=ia32" "--global-style" "--msvs_version=2015"
npm ERR! node v4.4.5
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE

npm ERR! atom-beautify@0.29.16 postinstall: `npm run build-options`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the atom-beautify@0.29.16 postinstall script 'npm run build-options'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the atom-beautify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build-options
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs atom-beautify
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls atom-beautify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\RODERI~1\AppData\Local\Temp\apm-install-dir-11702-6348-11bypu1\npm-debug.log
npm ERR! code 1
```

**Comment by albuic** [comment]
Hello,
I confirm the bug on win7 (Windows_NT 6.1.7601).
I have the same error log and also have a space in my name which seems to be the problem.
Happy new year ;-)

**Comment by Glavin001** [comment]
I think you may need to contact Atom team. See https://github.com/atom/apm/issues
I am not sure how `apm` is constructing and executing the arguments, they may need to escape spaces.

It looks like this may no longer be an issue with newer Atom? See https://github.com/atom/apm/issues/535#issuecomment-222390577

**Comment by afrz** [comment]
Thanks you for your response. I use lastest stable Atom version.
I created an issue in **apm** repository (see atom/apm#666). I will keep you posted.

But for @albuic and @rejhgadellaabsl it seems that atom/apm#659 will fix your problems (space in your username).

**Comment by pbakondy** [comment]
I have the same issue on Mac OSX

```
iMac:~ user$ atom --version
Atom    : 1.12.7
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0

iMac:~ user$ apm --version
apm  1.12.9
npm  3.10.5
node 4.4.5
python 2.7.13
git 2.11.0

iMac:~ user$ apm update --confirm=false
Package Updates Available (1)
└── atom-beautify 0.29.14 -> 0.29.16

Installing atom-beautify@0.29.16 to /Users/user/.atom/packages 
gyp info it worked if it ends with ok
gyp info using node-gyp@3.4.0
gyp info using node@4.4.5 | darwin | x64
gyp http GET https://atom.io/download/atom-shell/v1.3.13/iojs-v1.3.13.tar.gz
gyp WARN install got an error, rolling back install
gyp ERR! install error 
gyp ERR! stack Error: tunneling socket could not be established, cause=socket hang up
gyp ERR! stack     at ClientRequest.onError (/Applications/Atom.app/Contents/Resources/app/apm/node_modules/tunnel-agent/index.js:176:17)
gyp ERR! stack     at ClientRequest.g (events.js:260:16)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! stack     at ClientRequest.emit (events.js:169:7)
gyp ERR! stack     at Socket.socketOnEnd (_http_client.js:303:9)
gyp ERR! stack     at emitNone (events.js:72:20)
gyp ERR! stack     at Socket.emit (events.js:166:7)
gyp ERR! stack     at endReadableNT (_stream_readable.js:913:12)
gyp ERR! stack     at nextTickCallbackWith2Args (node.js:442:9)
gyp ERR! stack     at process._tickCallback (node.js:356:17)
gyp ERR! System Darwin 16.1.0
gyp ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/bin/node-gyp.js" "install" "--runtime=electron" "--target=1.3.13" "--dist-url=https://atom.io/download/atom-shell" "--arch=x64" "--ensure" "--proxy=http://192.168.150.11:8080/"
gyp ERR! cwd /Users/user/.atom
gyp ERR! node -v v4.4.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
```

**Comment by Glavin001** [comment]
Closing this issue as an `apm` bug, which I described reasoning in https://github.com/atom/apm/issues/666#issuecomment-270175194

**Comment by afrz** [comment]
For info, Atom version 1.14.1 fixed the problem.

**Comment by Glavin001** [comment]
I've merged https://github.com/Glavin001/atom-beautify/pull/1598 which should resolve these issues.

Published to v0.29.21.
